cirq/circuits/circuit_test.py

```diff
@@ -277,23 +310,26 @@ def test_add_iadd_equivalence():
     assert iadd_circuit == add_circuit
 
 
-def test_bool():
-    assert not cirq.Circuit()
-    assert cirq.Circuit(cirq.X(cirq.NamedQubit('a')))
+@pytest.mark.parametrize('frozen', [False, True])
+def test_bool(frozen):
+    assert not MaybeFrozenCircuit(frozen)
+    assert MaybeFrozenCircuit(frozen, cirq.X(cirq.NamedQubit('a')))
 
 
-def test_repr():
-    assert repr(cirq.Circuit()) == 'cirq.Circuit()'
+@pytest.mark.parametrize('frozen', [False, True])
+def test_repr(frozen):
+    type_name = 'FrozenCircuit' if frozen else 'Circuit'
```

Comment:
Here, could parametrize over the circuit class and name:
```python
@pytest.mark.parametrize('circuit_cls, type_name',
    [(cirq.Circuit, 'Circuit'), (cirq.FrozenCircuit, 'FrozenCircuit')])
def test_repr(circuit_cls, type_name):
    ...
```

Reply:
Opting to use `circuit_cls.__name__` instead.